packages/features/bookings/Booker/components/DatePicker.tsx

```diff
@@ -25,16 +29,30 @@ export const DatePicker = () => {
     <DatePickerComponent
       isLoading={schedule.isLoading}
       onChange={(date: Dayjs) => {
-        setSelectedDate(date.format("YYYY-MM-DD"));
+        if (syncChange) {
+          return updateQueryParam("date", date.format("YYYY-MM-DD"));
+        }
+        return replaceQueryParams("date", date.format("YYYY-MM-DD"));
       }}
       onMonthChange={(date: Dayjs) => {
-        setMonth(date.format("YYYY-MM"));
-        setSelectedDate(date.format("YYYY-MM-DD"));
+        if (typeof window === undefined) return;
+        const url = new URL(window.location.href);
+        url.searchParams.set("month", date.format("YYYY-MM"));
+        if (syncChange) {
+          if (date.isSame(dayjs(), "month")) {
+            url.searchParams.set("date", dayjs().format("YYYY-MM-DD"));
+          } else {
+            url.searchParams.set("date", date.format("YYYY-MM-DD"));
+          }
+        } else {
+          url.searchParams.delete("date");
+        }
+        router.push(url, url, { shallow: true });
```

Comment:
Should we not just move all this logic into the zustand store? so we keep all the onChange logic in one place?  Same goes for other places it has been changed

Reply:
It could be, but also I think that keeping the navigation status in Zustand is not necessary, the query params should be the state source.

Reply:
Yeah i agree with it being the source! Just think we should keep it in the store incase we need it elsewhere also. Looks good man nice job!